### ISSUE TITLE
Memory leak occurs when used proxy and the connection time out

### ISSUE THREAD

**Issue by hbasrc** [mojolicious/mojo#1767]
* Mojolicious version: ALL
* Perl version: 5.10+
* Operating system: Windows 6.1

### Steps to reproduce the behavior
When Mojo::UserAgent is used for frequent requests and http or socks proxy is set, if the target server service status is unstable and the connection timeout is frequent, the memory keeps growing.
I tracked the call stack and can confirm that there is a memory leak, but I don't know how to fix this problem.

**A simple example:**
```perl
#BEGIN { $ENV{MOJO_USERAGENT_DEBUG} = 1 }
use 5.010;
use strict;
use Mojo::UserAgent;
use Data::Printer;
#use Devel::TraceCalls;

#my $proxy = 'socks://127.0.0.1:2080';
my $proxy = 'http://127.0.0.1:2081'; # real proxy
my $url = 'https://10.0.0.1'; # dummy url for test

my $ua = Mojo::UserAgent->new;
#trace_calls $ua;
$ua->max_redirects(0)->connect_timeout(3)->request_timeout(6);
$ua->proxy->http($proxy)->https($proxy) if $proxy;

while (1) {
    get_url();
    sleep 2;
}

sub get_url {
    eval {
        my $res = $ua->get($url)->result;
        if ($res->is_success) {
            say $res->body;
        } else {
            say $res->message;
        }
    };

    if ($@) {
        print $@;
        p $ua->ioloop;
        #sleep 30;
    }
}

1;
```

**output:**
```
Mojo::IOLoop  {
.......omitted......
        out               {
            3c44ccbc0e597d1cc307b6467485a32a   {
                stream   Mojo::IOLoop::Stream
            },
            745d0d92369d6a3c3c0ecb87a6c8964a   {
                stream   Mojo::IOLoop::Stream
            },
            850d238f5f7e93ef7b54557d80fc2ca1   {
                stream   Mojo::IOLoop::Stream
            },
            .......omitted......
        },
        reactor           Mojo::Reactor::Poll
    }
}
```
### Expected behavior
The proxy connection session should be reused or removed

### Actual behavior
It's always increasing

**See the attachment log for the stack call**
[traceLog.txt](https://github.com/mojolicious/mojo/files/6328485/traceLog.txt)

**Comment by hbasrc** [comment]
Continue to track I found that the IO::Socket::SSL problem, and has been resolved in the v2.067, #https://github.com/noxxi/p5-io-socket-ssl/issues/92
As for what I said above is just a small problem, it can be solved by ioloop->reset. 

```perl
    if ($@) {
        print $@;
        $ua->ioloop->reset;
    }
```

**Comment by kraih** [comment]
Thanks for letting us know. Glad to see this fixed.
